docs/en/sql-reference/sql-functions/string-functions/append_trailing_char_if_absent.md

```diff
@@ -0,0 +1,58 @@
+---
+{
+    "title": "append_trailing_char_if_absent",
+    "language": "en"
+}
+---
+
+<!-- 
+Licensed to the Apache Software Foundation (ASF) under one
+or more contributor license agreements.  See the NOTICE file
+distributed with this work for additional information
+regarding copyright ownership.  The ASF licenses this file
+to you under the Apache License, Version 2.0 (the
+"License"); you may not use this file except in compliance
+with the License.  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing,
+software distributed under the License is distributed on an
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations
+under the License.
+-->
+
+# append_trailing_char_if_absent
+## description
+### Syntax
+
+`VARCHAR append_trailing_char_if_absent(VARCHAR str, VARCHAR trailing_char)`
+
+If the s string is non-empty and does not contain the c character at the end, it appends the c character to the end.
+Trailing_char contains only one character, and it will report an error if contains more than one character
```

Comment:
Why not return null when the second parameter length is not 1.
You can't check its length if it is not const, such as a table column.

Reply:
Because it's function name contains `trailing_char`, and this is also consistent with the clickhose function. I did a cast in fe `if (((StringLiteral) arg1).getValue().length() != 1) `, It will throw ClassCastException when SlotRef cast to StringLiteral. I can also add type checking if necessary

Reply:
In the case `append_trailing_char_if_absent(str, col1)` in which col1 is a column of a table. You can not know if the length of col1 is 1 or not.
Also there is no need to keep the same with CK.
It is convention to return null when the argument is invalid in Doris.

So what I suggest is to return null when the input argument is not valid. And there is no need to check its input in FE.

Reply:
OK, Done